Sources/ChatUI/ChatInChannel/MessageList.swift

```diff
@@ -76,22 +76,22 @@ public struct MessageList<MessageType: MessageProtocol & Identifiable, RowConten
                     isKeyboardShown = isShown
                 }
                 .coordinateSpace(name: listName)
-                .onPreferenceChange(ScrollViewOffsetPreferenceKey.self) { value in
-                    DispatchQueue.main.async { [self] in
-                        let offsetValue = value ?? 0
-                        let diff = offsetValue - scrollOffset
-                        scrollOffset = offsetValue
-                        
-                        let isScrollButtonShown = offsetValue < -20
-                        if showsScrollButton != isScrollButtonShown {
-                            showsScrollButton = isScrollButtonShown
-                        }
-                        
-                        if isKeyboardShown && diff < -20 {
-                            isKeyboardShown = false
-                        }
-                    }
-                }
+//                .onPreferenceChange(ScrollViewOffsetPreferenceKey.self) { value in
```

Comment:
This code handles the functionality for observing a significant user scroll then hiding the keyboard. With my app however it causes the keyboard to never be shown if I have scrolled (see the video attached to the pull request.

If I comment the code out then I don't have the problem of the keyboard not being shown after scroll.

Adding this pull request for discussion around whether you can see what the problem could be.

Cheers Fred :)